ci/scripts/python_wheel_windows_test.bat

```diff
@@ -35,7 +35,7 @@ set ARROW_TEST_DATA=C:\arrow\testing\data
 set PARQUET_TEST_DATA=C:\arrow\submodules\parquet-testing\data
 
 @REM Install testing dependencies
-pip install -r C:\arrow\python\requirements-wheel-test.txt || exit /B 1
+pip install tzdata -r C:\arrow\python\requirements-wheel-test.txt || exit /B 1
```

Comment:
Alternatively, we could also add it to `requirements-wheel-test.txt`? (with "environment markers", it should be possible to list a package only for windows, see https://peps.python.org/pep-0496/#examples)